crates/sui/tests/ptb_files_tests.rs

```diff
@@ -9,84 +9,85 @@ use sui::ptb::{
+    // === PREVIEW ===
+    let ptb_preview = ptb.preview(&commands);
+    let mut results = vec![];
+    let preview_string = if let Some(ptb_preview) = ptb_preview {
+        ptb_preview.to_string()
+    } else {
+        "".to_string()
+    };
+    results.push(" === PREVIEW === ".to_string());
+    results.push(preview_string);
 
-    Ok(())
-}
-
-// check if the bad ptb files return errors when parsed
-fn test_parse_bad_ptb_files(path: &Path) -> datatest_stable::Result<()> {
-    let ptb = PTB::default();
-    let cmd = PTB::command();
-    let file = path.to_str().unwrap();
-    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
-    let cwd = std::env::current_dir().unwrap();
-    let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();
+    // === PARSE COMMANDS ===
     let mut parser = PTBParser::new();
-    for (_, cmd) in commands.clone() {
+    for (_, cmd) in &commands {
         parser.parse(cmd.clone());
     }
-    let (_parsed_commands, errors) = parser.finish();
-    let rendered = render_errors(commands, errors.clone());
-    for e in rendered.iter() {
-        println!("{:?}", e);
+    let (parsed, errors) = parser.finish();
+
+    results.push(" === Parsed PTB === ".to_string());
+    for c in &parsed {
+        let values = c
+            .args
+            .iter()
+            .map(|x| x.value.to_string())
+            .collect::<Vec<_>>();
+        results.push(format!(
+            "cmd: {}, value: {:?}",
+            c.name.value.to_string(),
+            values
+        ));
     }
-    println!("{:?}", errors.len());
-    assert!(!errors.is_empty());
-    Ok(())
-}
 
-// parse, build PTB, and run it locally in a network
-async fn test_build_and_run_ptb_from_files(path: &Path) -> datatest_stable::Result<()> {
-    let ptb = PTB::default();
-    let cmd = PTB::command();
-    let file = path.to_str().unwrap();
-    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
+    // === BUILD PTB ===
     let test_cluster = TestClusterBuilder::new().build().await;
     let context = test_cluster.wallet;
+    let client = context.get_client().await?;
 
-    ptb.execute(args, Some(context)).await.unwrap();
-    Ok(())
-}
+    let built_ptb = ptb.parse_and_build_ptb(parsed, &context, client).await;
 
-async fn test_no_gas_picker_from_files(path: &Path) -> datatest_stable::Result<()> {
-    let ptb = PTB::default();
-    let cmd = PTB::command();
-    let file = path.to_str().unwrap();
-    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
-    let test_cluster = TestClusterBuilder::new().build().await;
-    let context = test_cluster.wallet;
+    if let Ok(ref ptb) = built_ptb {
+        results.push(" === BUILT PTB === ".to_string());
+        for c in &ptb.0.commands {
+            results.push(c.to_string());
+        }
+    }
+
+    // === ERRORS ===
+    let mut errors = errors.clone();
+    if let Err(e) = built_ptb {
+        errors.extend(e.clone());
+    }
```

Comment:
IMO I would probably not try to build the PTB if we encounter parsing errors since that may lead to a bunch of weird errors that you wouldn't get normally (since we stop if we encounter parsing errors in the CLI).